test/tests_dag.py

```diff
@@ -0,0 +1,368 @@
+        "AI.TENSORSET persisted_tensor_1 FLOAT 1 2 VALUES 5 10")
+    env.assertEqual(ret, b'OK')
+    command = "AI.DAGRUN "\
+                "LOAD 1 persisted_tensor_1 "\
+                "PERSIST 1 tensor1 |> "\
+              "AI.TENSORSET tensor1 FLOAT 1 2 VALUES 5 10"
+
+    ret = con.execute_command(command)
+    env.assertEqual(ret, [b'OK'])
+
+
+def test_dag_local_tensorset(env):
+    con = env.getConnection()
+
+    command = "AI.DAGRUN "\
+        "AI.TENSORSET volative_tensor1 FLOAT 1 2 VALUES 5 10 |> "\
+        "AI.TENSORSET volative_tensor2 FLOAT 1 2 VALUES 5 10 "
+
+    ret = con.execute_command(command)
+    env.assertEqual(ret, [b'OK',b'OK'])
+
+    # assert that transaction tensor does not exist
+    ret = con.execute_command("EXISTS volative_tensor")
+    env.assertEqual(ret, 0 )
+
+def test_dag_local_tensorset_persist(env):
+    con = env.getConnection()
+
+    command = "AI.DAGRUN "\
+        "PERSIST 1 tensor1 |> "\
+        "AI.TENSORSET tensor1 FLOAT 1 2 VALUES 5 10"
+
+    ret = con.execute_command(command)
+    env.assertEqual(ret, [b'OK'])
+
+    # assert that transaction tensor exists
+    ret = con.execute_command("EXISTS tensor1")
+    env.assertEqual(ret, 1 )
+
+    ret = con.execute_command("AI.TENSORGET tensor1 VALUES")
+    env.assertEqual(ret, [b'FLOAT', [1, 2], [b'5', b'10']])
+
+
+def test_dag_multilocal_tensorset_persist(env):
+    con = env.getConnection()
+
+    command = "AI.DAGRUN "\
+        "PERSIST 1 tensor3 |> "\
+        "AI.TENSORSET tensor1 FLOAT 1 2 VALUES 5 10 |> "\
+        "AI.TENSORSET tensor2 FLOAT 1 2 VALUES 5 10 |> "\
+        "AI.TENSORSET tensor3 FLOAT 1 2 VALUES 5 10 |> "\
+        "AI.TENSORSET tensor4 FLOAT 1 2 VALUES 5 10 "
+
+    ret = con.execute_command(command)
+    env.assertEqual([b'OK',b'OK',b'OK',b'OK'],ret)
+
+    # assert that transaction tensor exists
+    ret = con.execute_command("EXISTS tensor1")
+    env.assertEqual(ret, 0 )
+
+    # assert that transaction tensor exists
+    ret = con.execute_command("EXISTS tensor2")
+    env.assertEqual(ret, 0 )
+
+    # assert that transaction tensor exists
+    ret = con.execute_command("EXISTS tensor3")
+    env.assertEqual(ret, 1 )
+
+    # assert that transaction tensor exists
+    ret = con.execute_command("EXISTS tensor4")
+    env.assertEqual(ret, 0 )
+
+    ret = con.execute_command("AI.TENSORGET tensor3 VALUES")
+    env.assertEqual(ret, [b'FLOAT', [1, 2], [b'5', b'10']])
+
+def test_dag_local_tensorset_tensorget_persist(env):
+    con = env.getConnection()
+
+    command = "AI.DAGRUN PERSIST 1 tensor1 |> "\
+        "AI.TENSORSET tensor1 FLOAT 1 2 VALUES 5 10 |> "\
+        "AI.TENSORGET tensor1 VALUES"
+
+    ret = con.execute_command(command)
+    env.assertEqual(ret, [b'OK', [b'FLOAT', [1, 2], [b'5', b'10']]])
+
+    ret = con.execute_command("AI.TENSORGET tensor1 VALUES")
+    env.assertEqual(ret, [b'FLOAT', [1, 2], [b'5', b'10']])
+
+
+def test_dag_local_multiple_tensorset_on_same_tensor(env):
```

Comment:
> We need to be careful with the semantics of values we reply with from the local context. Say we have:
> 
> ```
> TENSORSET foo ...
> MODELRUN m1 INPUTS foo OUTPUTS bar
> TENSORGET bar
> MODELRUN m2 INPUTS foo OUTPUTS bar
> ```
> 
> then we should get `bar` after `m1` has run, not `m2`.
> 
> Ok, this uses two MODELRUN's, but in the current code it's the same thing as
> 
> ```
> TENSORSET foo ...
> TENSORSET bar ...
> TENSORGET bar ...
> MODELRUN m INPUTS foo OUTPUTS bar
> TENSORGET bar ...
> ```
> 
> I would expect `bar` to have two different values in output. What do you think?

@lantiga agree and we have it working with the current implementation. Please revise the `test_dag_local_multiple_tensorset_on_same_tensor` added test